.github/workflows/test.yml

```diff
@@ -0,0 +1,52 @@
+name: test
+on: 
+  push:
+    branches-ignore:
+      - 'master'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Environment Config
+      run: |
+        echo "$HOME/go/bin" >> $GITHUB_PATH
+        # Set /bin writable for golangci-lint.sh (TODO: refactor w/ golangci/golangci-lint-action@v2 action)
+        sudo chmod 777 /bin/
+    - name: Checkout Repo
+      uses: actions/checkout@v2
+    - name: Restore Go Modules Cache
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-go-modules
+      with:
+        path: /go/pkg/mod
+        key: v1-dependencies-{{ checksum "go.sum" }}
+    - name: Git Config
+      run: |
+        git config --global user.email circleci
+        git config --global user.name circleci
+    - name: Install Goreleaser Binary
+      run: |
+        curl -sL https://install.goreleaser.com/github.com/goreleaser/goreleaser.sh | bash
+        sudo mv ./bin/goreleaser /bin
+    - name: Install Coverage Tools
+      run: |
+        make deps-coverage
+    - name: Retrieve Modules
+      run: |
+        go get
+    - name: Run Coverage Report
+      run: |
+        make test-circle
+    - name: Build Autotag
+      run: |
+        make build
+    - name: Goreleaser Check
+      run: |
+        goreleaser check
+    - name: Upload Autotag Build
+      uses: actions/upload-artifact@v2
+      with:
+        name: autotag-build
+        path: .
```

Comment:
Looks like a duplication of `.github/workflows/release.yml` file.
What about having one yml and adding some check for the `release` step, like it's not a PR e.g.: https://stackoverflow.com/a/64210623